test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+            selection: d3.select(barPlot.content().selectAll("rect")[0][index]),
+            component: barPlot
+          };
+        }
+
+        describe("retrieving the nearest Entity", () => {
+          function testEntityNearest() {
+            data.forEach((datum, index) => {
+              const expectedEntity = expectedEntityForIndex(index);
+
+              const barBasePosition = baseScale.scale(datum.base);
+
+              const halfwayValuePosition = valueScale.scale((barPlot.baselineValue() + datum.value) / 2);
+              const pointInsideBar = getPointFromBaseAndValuePositions(barBasePosition, halfwayValuePosition);
+              const nearestInsideBar = barPlot.entityNearest(pointInsideBar);
+              TestMethods.assertPlotEntitiesEqual(nearestInsideBar, expectedEntity, "retrieves the Entity for a bar if inside the bar");
+
+              const abovePosition = valueScale.scale(2 * datum.value);
+              const pointAboveBar = getPointFromBaseAndValuePositions(barBasePosition, abovePosition);
+              const nearestAboveBar = barPlot.entityNearest(pointAboveBar);
+              TestMethods.assertPlotEntitiesEqual(nearestAboveBar, expectedEntity,
+                "retrieves the Entity for a bar if beyond the end of the bar");
+
+              const belowPosition = valueScale.scale(-datum.value);
+              const pointBelowBar = getPointFromBaseAndValuePositions(barBasePosition, belowPosition);
+              const nearestBelowBar = barPlot.entityNearest(pointBelowBar);
+              TestMethods.assertPlotEntitiesEqual(nearestBelowBar, expectedEntity,
+                "retrieves the Entity for a bar if on the other side of the baseline from the bar");
+            });
+          }
+
+          it("returns the closest by base, then by value", () => {
+            testEntityNearest();
+          });
+
+          it("returns the closest visible bar", () => {
+            const baseValuebetweenBars = (data[0].base + data[1].base) / 2;
+            baseScale.domain([baseValuebetweenBars , DEFAULT_DOMAIN[1]]); // cuts off bar 0
+            const bar0BasePosition = baseScale.scale(data[0].base);
+            const baselineValuePosition = valueScale.scale(barPlot.baselineValue());
+
+            const nearestEntity = barPlot.entityNearest(getPointFromBaseAndValuePositions(bar0BasePosition, baselineValuePosition));
+            const expectedEntity = expectedEntityForIndex(1); // nearest visible bar
+            TestMethods.assertPlotEntitiesEqual(nearestEntity, expectedEntity, "returned Entity for nearest in-view bar");
+          });
+
+          it("considers bars cut off by the value scale", () => {
+            valueScale.domain([-0.5, 0.5]);
+            testEntityNearest();
+          });
+
+          it("considers bars cut off by the base scale", () => {
+            baseScale.domain([-0.8, 0.8]);
+            testEntityNearest();
+          });
```

Comment:
Cuts the bars off partially since their positions are at `-1` and `1`. Again, possible to programmatically generate the domain, but it might be hard to follow.